openrelik_worker_common/utils.py

```diff
@@ -166,48 +142,60 @@ def to_dict(self):
             A dictionary containing the attributes of the OutputFile object.
         """
         return {
+            "uuid": self.uuid,
             "display_name": self.display_name,
+            "extension": self.extension,
             "data_type": self.data_type,
-            "uuid": self.uuid,
             "path": self.path,
             "original_path": self.original_path,
             "source_file_id": self.source_file_id,
         }
 
 
 def create_output_file(
-    output_path: str,
-    filename: Optional[str] = None,
-    uuid_as_filename: Optional[bool] = False,
-    add_extension: Optional[str] = None,
-    data_type: Optional[str] = "openrelik:worker:file:generic",
+    output_base_path: str,
+    display_name: Optional[str] = None,
+    extension: Optional[str] = None,
+    data_type: Optional[str] = None,
     original_path: Optional[str] = None,
     source_file_id: Optional[OutputFile] = None,
 ) -> OutputFile:
     """Creates and returns an OutputFile object.
 
     Args:
-        output_path: The path to the output directory.
-        filename: The name of the output file (optional).
-        uuid_as_filename: Use UUID as filename (optional).
-        add_extension: File extension (optional).
+        output_base_path: The path to the output directory.
+        display_name: The name of the output file (optional).
+        extension: File extension (optional).
         data_type: The data type of the output file (optional).
```

Comment:
I think here it would make sense to set the default value to "openrelik:worker:file:generic" like we did before.

Reply:
This has changed. The default data_type will be set on the server and uses the mime type as as a base.
e.g: `file:application:x-ms-evtx`